flowToScript.js

```diff
@@ -1,55 +1,126 @@
+      response = res;
+      res.on("data", (data) => {
+        process.stdout.write(data);
+      });
+    });
+  
+    req.on("error", (e) => {
+      console.error(e);
+    });
+    req.end();
+    return response;
+    };
+    requestBlock();`;
+    if (document.getElementById('apiKey').value) {
+      makeRequestString = makeRequestString.replace(
+        'null',
+        document.getElementById('apiKey').value
+      );
+    }
+  } else {
+    makeRequestString = 'Base URL not found. Try importing a swagger file.';
+  }
+  try {
+    evaluateBlock(0);
+    requestString += '}; \n';
+  } catch {
+    requestString = 'Script not found. Try adding blocks or parameters.';
+  }
 }
 
 function evaluateBlock(id) {
   let block = getBlock(id);
+  let children = getChildBlocks(id);
+  let startI = 0;
   //If it's an API block, do a thing
-
   if (getDataProperty(block['data'], 'method')) {
-    //
+    blockUrl = block['data'][2].value.substring(baseUrl.lastIndexOf(':') + 3);
+    requestString += `resp = await makeRequest("${blockUrl.replace(
+      /\/$/,
+      ''
+    )}", ${JSON.stringify(block['data'][0].value)},"`;
+    requestString += JSON.stringify(block['data'][1].value);
+    props = chartProperties[id]['properties'];
+    let queryArr = [];
+    let bodyString = '';
+    for (prop of props) {
+      if (prop.value == undefined) {
+        continue;
+      }
+      if (prop.in == 'path') {
+        requestString = requestString.replace(
+          `{${prop.name}}`,
+          `${prop.value}`
+        );
+      } else if (prop.in == 'query') {
+        queryArr.push(`${prop.name}=${prop.value}`);
+      } else if (prop.in == 'body') {
+        bodyString += `, ${prop.value}`;
+      }
+    }
+    if (queryArr[0]) {
+      requestString = requestString.substring(0, requestString.length - 1);
+      requestString += '?' + queryArr.join('&') + '"';
+    }
+    requestString += bodyString;
+    requestString += ');';
   } else if (getDataProperty(block['data'], 'logic')) {
-    //Handle if or for blocks
-    //console.log("Logic");
+    paramValue = chartProperties[id]['properties'][0].value;
+    if (block['data'][0].value == 'if') {
+      requestString += `if(${paramValue}){`;
+      id++;
+      evaluateBlock(id);
+      requestString += `}else{`;
+      id++;
+      evaluateBlock(id);
+      requestString += `}`;
+      startI += 2;
+    } else if (block['data'][0].value == 'for') {
+      requestString += `for(${paramValue}){`;
+      id++;
+      evaluateBlock(id);
+      requestString += `}`;
+      startI++;
+    } else if (block['data'][0].value == 'log') {
+      requestString += `console.log(${paramValue});`;
+    } else if (block['data'][0].value == 'set') {
+      requestString += `let ${paramValue} = ${chartProperties[id]['properties'][1].value};`;
+    }
   }
 
-  let children = getChildBlocks(id);
-  for (let i = 0; i < children.length; i++) {
+  requestString += '\n';
+
+  for (let i = startI; i < children.length; i++) {
```

Comment:
do we need to `return` inside the if/else blocks above to avoid calling evaluateBlock multiple times for the same children?

Reply:
we don't appear to since I increment the starting id value for the children

Reply:
but in that case, you're evaluating blocks after the `if` so if your if block is 3, you're calling `evaluateBlock()` for 2, 3, 4, and 5

Reply:
nevermind. I see what you're doing. Seems confusing to keep track of the start index rather than just always doing `i = 0` and then returning when needed above